src/DotNetLightning.Core/Transactions/Transactions.fs

```diff
@@ -735,19 +736,29 @@ module Transactions =
                 else
                     spec.ToLocal.ToMoney(), spec.ToRemote.ToMoney() - closingFee
 
-            let maybeToLocalOutput = if (toLocalAmount >= dustLimit) then Some(toLocalAmount, localDestination) else None
-            let maybeToRemoteOutput = if (toRemoteAmount >= dustLimit) then Some(toRemoteAmount, remoteDestination) else None
-            let outputs = seq [maybeToLocalOutput; maybeToRemoteOutput] |> Seq.choose id
+            let maybeToLocalOutput = if (toLocalAmount >= dustLimit) then Some(TxOut(toLocalAmount, localDestination)) else None
+            let maybeToRemoteOutput = if (toRemoteAmount >= dustLimit) then Some(TxOut(toRemoteAmount, remoteDestination)) else None
+            let outputs =
+                seq [maybeToLocalOutput; maybeToRemoteOutput]
+                |> Seq.choose id
+                |> Seq.sortBy (fun txOut ->
+                    use memoryStream = new MemoryStream()
+                    let stream = NBitcoin.BitcoinStream(memoryStream, true)
+                    txOut.ReadWrite stream
+                    memoryStream.ToArray()
+                )
             let psbt = 
-                let txb = n.CreateTransactionBuilder()
-                           .AddCoins(commitTxInput)
-                           .SendFees(closingFee)
-                           .SetLockTime(!> 0u)
-                for (money, dest) in outputs do
-                    txb.Send(dest, money) |> ignore
-                let tx =  txb.BuildTransaction(false)
+                let txIn =
+                    let txIn = TxIn()
+                    txIn.Sequence <- !> 0xffffffffu
+                    txIn.PrevOut <- commitTxInput.Outpoint
```

Comment:
rather initialize it without mutation `let txIn = TxIn(PrevOut = commitTxInput.Outpoint, Sequence = !> 0xffffffffu)`

Reply:
also put 0xffffffffu under a constant (so that we know what it refers to)